### ISSUE TITLE
Extra closing curly brace is added with wrapped at-rules and stylis-rule-sheet

### ISSUE THREAD

**Issue by kitten** [thysultan/stylis.js#99]
Not sure if this is something with our implementation in styled-components, but as described in the issue here (https://github.com/styled-components/styled-components/issues/1576#issuecomment-370818961) invalid output can be generated when nesting media at-rules inside normal style rules.

I've extracted a reproduction into a CodeSandbox here: https://codesandbox.io/s/8kop505kjl

I'm not entirely convinced that this only happens with this certain input.

**Comment by thysultan** [comment]
Thanks, from the looks of it, commenting out.

```js
stylisWithPlugin.use([parseRulesPlugin, returnRulesPlugin]);
```

reveals that it's coming from the stylis-rule-sheet plugin.

**Comment by kitten** [comment]
It does seem to work when I replace `(depth === 0 ? delimiter : '')` with just `delimiter`. Not sure what exception the `depth !== 0` is addressing, or what effect removing this exception has.

https://github.com/thysultan/stylis.js/blob/master/plugins/rule-sheet/index.js#L41

**Comment by thysultan** [comment]
Related [commit](https://github.com/thysultan/stylis.js/commit/93936083f52e841395ba280846136c34fef3c241#diff-c2f28f1c4ff12ca126040710834fbeaf)

Which is related to https://github.com/thysultan/stylis.js/pull/93

I'll see if i can find another fix for that.

**Comment by thysultan** [comment]
Fixed in stylis-rules-heet@0.0.10 & stylis@3.5.0, related [changelog entry](https://github.com/thysultan/stylis.js/blob/master/CHANGELOG.md#350-march-08-2018)
